pkg/tfbridge/names.go

```diff
@@ -28,6 +28,15 @@ import (
 // PulumiToTerraformName performs a standard transformation on the given name string, from Pulumi's PascalCasing or
 // camelCasing, to Terraform's underscore_casing.
 func PulumiToTerraformName(name string, tfs shim.SchemaMap, ps map[string]*SchemaInfo) string {
+
+	if tfs != nil {
```

Comment:
One question I have is do we always have a complete set of TF names passed in `tfs shim.SchemaMap`, or are there situations where we legitimately are trying to "make up a name" on the fly that's not part of the schema, or else we're just forgetting to pass the complete set of TF names?

If we can make this stricter and we are certain we have complete name information available, then the fallback name computation should probably be removed, and this function deprecated into a version that returns an `(string, error)` and errors out if the TF name is not found in the table.